lib/model.js

```diff
@@ -3085,14 +3086,23 @@ class Model {
 
             // Convert date fields to real date objects
             if (this.constructor._hasDateAttributes && this.constructor._isDateAttribute(key) && !!value && !value._isSequelizeMethod) {
-              if (!(value instanceof Date)) {
-                value = new Date(value);
-              }
-              if (!(originalValue instanceof Date)) {
-                originalValue = new Date(originalValue);
-              }
-              if (originalValue && value.getTime() === originalValue.getTime()) {
-                return this;
+              // Dont parse DATEONLY to new Date, keep them as string
+              if (this.rawAttributes[key].type instanceof DataTypes.DATEONLY) {
+                if (originalValue && originalValue === value) {
+                  return this;
+                } else {
+                  value = moment(value).format('YYYY-MM-DD');
```

Comment:
` Object.assign(new User(), {birthday: new Date()});` Syntax wont work actually as we hide data in `dataValues`

Beside we are already converting incoming user data to `Date` instance as well. So I dont think there should be any problem converting it to `YYYY-MM-DD`, because thats what `DATEONLY` field is storing